EKS-POD-IDENTITY-README.md

```diff
@@ -0,0 +1,160 @@
+permit you to make these changes without proper approvals in place adding more pressures on the operational process. 
+
+EKS Pod Identities are introduced by AWS to address this operational challenge
+
+### EKS Pod Identities
+
+
+What makes EKS Pod Identities impervious to the challenge of changing EKS clusters?  The key is understanding who
+the underlying Principal is in the trust policy.
+
+In IRSA your trust policy for the IAM Role looks something like this
+```json
+{
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "$PROVIDER_ARN"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${ISSUER_HOSTPATH}:sub": "system:serviceaccount:default:my-serviceaccount"
+        }
+      }
+    }
+  ]
+}
+```
+In the above the `$PROVIDER_ARN` is the OIDC endpoint associated with the EKS cluster. And the `${ISSUER_HOSTPATH}:sub`
+has to include every K8s Service Account that is permitted to assume this role. Change the EKS cluster and the former 
+changes. If too many users can assume the role, you run the risk of hitting the (extended) limit of 4096 characters 
+for the trust policy.
+
+Compare this with the trust policy for an IAM role when using EKS Pod Identity
+
+```json
+"Principal": {
+                "Service": "pods.eks.amazonaws.com"
+            }
+```
+There is an additional configuration needed. The `AmazonEKSWorkerNodePolicy` which is attached to the node role 
+`<cluster-name>-eks-nodes` must contain the statement below.
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "WorkerNodePermissions",
+            "Effect": "Allow",
+            "Action": [
+                "eks-auth:AssumeRoleForPodIdentity"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+The above policy is too permissive and allows the Node Policy to assume all IAM Roles. Expect yours to be more restrictive
+such that it allows Domino Workloads to assume specific AWS IAM Roles.
+
+Notice how there are no references to the OIDC endpoint or the actual K8s accounts mapped to the role. This mapping 
+happens in the EKS cluster settings itself
+```shell
+aws eks create-pod-identity-association --cluster-name my-cluster --role-arn arn:aws:iam::111122223333:role/my-role --namespace default --service-account my-service-account
+```
+
+### Bringing it together
+
+Now imagine doing a Blue/Green upgrade in Domino. When you create the new (Green) EKS cluster, you copy these mappings
+into the new cluster. No IAM ownership is needed. The owner of the EKS cluster could perform this action.
+
+When you retire the Blue EKS cluster, simply delete the cluster and your mappings get destroyed with the cluster.
+
+### Comparison between EKS Pod Identities and IRSA
+
+
+The main challenge with Pod Identities is, you cannot switch roles inside a workload. These mappings are static and 
```

Comment:
Could we put this into a table so it's easier to gauge the pros/cons?

Reply:
Done